content/livesync/models.textile

```diff
@@ -0,0 +1,354 @@
+
+h3(#replay-from-history). Replay from history
+
+The SDK will initialise model state using the sync function, and capture the @sequenceId@ you return from your sync function. The SDK then connects to the model's channel, and will rewind backwards though history until it finds the correct point in the stream of channel messages (based on the sequenceId). You can configure how many messages are requested in each page of history when rewinding using the parameter @syncOptions.historyPageSize@ defined in your @ClientOptions@.
+
+After the SDK executes the sync function, it internally connects to the "channel":/channels and begins "subscribing":/channels to live, yet unprocessed, messages.  It then paginates backward through the channel's message history, starting from the attachment point, using "@untilAttach@":/storage-history/history#until-attach. The number of items on each page can be configured via the @syncOptions.historyPageSize@ defined in your "@ClientOptions@":#client-options.
+
+When a message is published to the channel with a new @sequenceId@, the model processes subsequent messages to maintain continuity with live messages. If the target @sequenceId@ is unreachable, there is insufficient message history to resume from the correct point to update the model. The SDK will attempt a re-sync. It re-syncs by calling the sync function again to acquire a newer version of the model state.
+
+The amount of history available to query on the channel is determined by your "message storage":/storage-history/storage configuration on the channel. This configuration must match the @syncOptions.messageRetentionPeriod@ defined in your "@ClientOptions@":#client-options. The SDK uses this configuration option as a hint as to whether to resynchronize via the sync function and skip paginating through history when messages are expected to have expired.
+
+To ensure the model's state is up to date, cache the state obtained from the backend endpoint used by the sync function. The model state returned from this endpoint must not exceed the message retention period set on the channel. If no previous messages exist on the channel, the model is assumed to be a new state without any modifications.
+
+h2(#create-merge). Create a Merge function
+
+A merge function is required when you instantiate a model.
+
+When a message is received on the channel, the merge function is called to merge that new message into the existing Model state. The merge function is called with the existing Model state and the newly received channel message, and it should return the updated Model state (with the new message merged in).
+
+It's important that the merge function is pure and deterministic, and not depending on any external state. This is because the merge will often be called multiple times and in different orders (for example when replaying events after rewinding to a sequenceId, or when rebasing optimistic events on top of confirmed events).
```

Comment:
```suggestion
It's important that the merge function is pure and deterministic, and doesn't depend on any external state. This is because the merge will often be called multiple times and in different orders (for example when replaying events after rewinding to a @sequenceId@, or when rebasing optimistic events on top of confirmed events).
```